### PR TITLE
Issue #20349 - Categorize unused tabs into an inactive group in the tabs tray

### PR DESCRIPTION
**Pull request by jonalmeida**
~Playing around with an idea of piping the TabsFeature to an intermediate `TabsTray` (ActiveTabSeparator) that then pipes it out to two `RecyclerView.Adapter`, which are used on one RecyclerView via a `ConcatAdapter`.~

@rocketsroger came up with an excellent suggestion instead to use two `TabsFeature` for active/inactive adapters.

Since we already split up the adapters for normal/private tabs, there is enough of a diversion in our layouts, that we needed to split up the tray lists (RecyclerView) as well.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayController.kt [hunk 1]
```diff
@@ -53,6 +58,16 @@ interface TabsTrayController {
...
     fun handleMultipleTabsDeletion(tabs: Collection<Tab>)
+
+    /**
+     * Set the list of [tabs] into the inactive state.
+     *
+     * @param tabs List of [Tab]s to be removed.
+     */
+    fun forceTabsAsInactive(
+        tabs: Collection<Tab>,
+        numOfDays: Long = DEFAULT_INACTIVE_TEST_DAYS
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayController.kt:69]:
```suggestion
        numOfDays: Long = DEFAULT_INACTIVE_DAYS + 1
```

**Review by jonalmeida — COMMENTED** [review]
A review from @rocketsroger while pairing:

**Comment by jonalmeida** [comment]
@rocketsroger this is ready for review.

@codrut-topliceanu I added you as a reviewer as well since this was blocking https://github.com/mozilla-mobile/fenix/issues/20310.

File: app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayController.kt [hunk 2]
```diff
@@ -53,6 +57,18 @@ interface TabsTrayController {
...
     fun handleMultipleTabsDeletion(tabs: Collection<Tab>)
+
+    /**
+     * Set the list of [tabs] into the inactive state.
+     *
+     * ⚠️ DO NOT USE THIS OUTSIDE OF DEBUGGING/TESTING.
+     *
+     * @param tabs List of [Tab]s to be removed.
+     */
+    fun forceTabsAsInactive(
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayController.kt:68]: Nice!

**Review by rocketsroger — APPROVED** [review]
🚢

**Comment by pocmo** [comment]
Rebasing manually for merging manually (mergify queue is broken)

**Pull request merged**